cranelift/codegen/src/isa/riscv64/lower.isle

```diff
@@ -710,6 +710,64 @@
   (if-let x_imm (replicated_imm5 x))
   (rv_vand_vi y x_imm (unmasked) ty))
 
+;; `bclr{,i}` specializations from `zbs`
+
+(rule 13 (lower (has_type $I32 (band x (bnot (ishl (i64_from_iconst 1) y)))))
+  (if-let $true (has_zbs))
+  (rv_bclr x (rv_andi y (imm12_const 31))))
```

Comment:
A small suggestion here would be to change this to `(u64_sub (ty_bits ty) 1)` so that we could also apply these patterns to i16 and i8.

Reply:
Also, perhaps we should add equivalent rules for these with swapped arguments. i.e. `(band (bnot ...) y)` we have them for the other  instructions but not for `bclr`.